### DISCUSSION TITLE
Documentation: MP3 conversion encoding quality

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [openaudible/openaudible#948] [Q&A]
First I would like to say: great work, I love your software and the possibilities it gives me. 

Now the question: I couldn't find anywhere the parameters used for the various encoding quality steps in the MP3 conversion setting pull-down (well, apart from going through the code, that I didn't do). Is there any list of them? 

Why do I ask: the recommended quality produces a very satisfactory results - with the exception of my car audio system. Unfortunately, it doesn't really support the VBR MP3 files, which is exactly what this setting produces. Could you tell me, which setting produces CBR MP3 files? I don't really need to do anything with the bitrate or the joint-stereo option, but I need CBR.
Thanks!

**Comment by Augustinus-Altovadensis** [comment]
OK, I went into the code, so I will start with the reply, but I still need some help, so please bear on :-) 

So, the file `/openaudible/convert/ConvertJob.java` says on line 145+:
`		args.add("`-qscale:a`");`      // https://trac.ffmpeg.org/wiki/Encode/MP3
		
		
		if (mp3Qscale < 0 || mp3Qscale > 9)
			throw new IOException("Invalid qscale:" + mp3Qscale);
		
		args.add("" + mp3Qscale);`

So I went to the suggested page, and it says that this way, FFmpeg will create _only_ VBR, in order to change it to CBR, one would need to change the argument from `-qscale:a` to say `-b:a 256k`, so there is no way to change it in the current code. The solution is either to add the option, or for me to change this one argument in the java file and recompile? I don't really know my way around java programs, I have written quite a few things in c, pascal and perl, but nothing in java... I guess I could get to the bottom of this, but in order to keep with the future development, it would be more practical, if you added the option into the program settings. Thanks for any suggestions!

**Comment by Cfomodz** [comment]
Edit `src/main/java/org/openaudible/convert/ConvertJob.java` (lines 137-143):

**Replace:**
```java
args.add("-qscale:a");
if (mp3Qscale < 0 || mp3Qscale > 9)
    throw new IOException("Invalid qscale:" + mp3Qscale);
args.add("" + mp3Qscale);
```

**With:**
```java
args.add("-b:a");
args.add("128k");  // CBR - change to 64k/96k/192k/256k as needed
```

Recompile OpenAudible.

## Post-Process Alternative

Convert existing VBR files to CBR:
```bash
ffmpeg -i input.mp3 -codec:a libmp3lame -b:a 128k output.mp3
```
